eras/conway/impl/testlib/Test/Cardano/Ledger/Conway/CDDL.hs

```diff
@@ -853,3 +853,11 @@ nonempty_set = binding $ \x ->
   "nonempty_set"
     =:= tag 258 (arr [1 <+ a x])
     / sarr [1 <+ a x]
+
+-- | An OSet is a Set that preserves the order of its elements.
+oset :: IsType0 t0 => t0 -> GRuleCall
+oset = binding $ \x -> "oset" =:= tag 258 (arr [0 <+ a x])
```

Comment:
I'd suggest extracting definition of set specification and using it for 

```suggestion
oset = binding $ \x -> "oset" =:= set_len_spec 0

set_len_spec n = tag 258 (arr [n <+ a x]) / sarr [n <+ a x]
```

and the same for nonempty and regular set.